File: src/tmp/Python/refer-offer/er_cha_shu_de_jing_xiang_lcof.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def mirrorTree(root: TreeNode) -> TreeNode:
    def change(node: TreeNode) -> TreeNode:
        if not node or (not node.left and not node.right):
            return node
        
        temp = node.left
        node.left = change(node.right)
        node.right = change(temp)

        return node
    
    return change(root)

File: core/tmp/Java/src/com/dbc/algorithm/ReferOffer/ErChaShuDeJingXiangLcof.java
package com.dbc.algorithm.ReferOffer;

public class ErChaShuDeJingXiangLcof {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    public TreeNode mirrorTree(TreeNode root) {
        if (root == null || (root.left == null && root.right == null)) {
            return root;
        }
        TreeNode temp = root.left;
        root.left = mirrorTree(root.right);
        root.right = mirrorTree(temp);

        return root;
    }
}
